### ISSUE TITLE
client_PublishArrived Issue with GnatMQ

### ISSUE THREAD

**Issue by JayEnaR** [stevenlovegrove/MqttDotNet#14]
![capture4](https://cloud.githubusercontent.com/assets/25684062/22860063/590ca5a8-f0fa-11e6-9c6d-9b23e89440bb.PNG)
Everything else seems to work fine but as soon as an incoming message is received (either from a direct  or a retained message) I get an error stipulated in the log [here](https://docs.google.com/document/d/1R9jc7yxuegF3pq7pbDo8YudTBH-mH6VQ376zwk0hAuc/edit?usp=sharing) and the attached screenshot.

I'm using the default [MqttDotNet Sample project](https://github.com/stevenlovegrove/MqttDotNet) and libraries to connect to GnatMQ from @ppatierno as a Broker.

Basically it seems like the issue is caught in QosManager.cs at line 91:
`else if (mess.QualityOfService == QoS.OnceAndOnceOnly)
              {
                _responses.Add(mess.MessageID, new MqttPubrelMessage(mess.MessageID));
              }`

Anyone know where the issue could be coming from?

**Comment by darkstarlabs** [comment]
This is not localised to GnatMQ as mosquitto generates the same problem.
